src/sentry/spans/consumers/process/factory.py

```diff
@@ -40,16 +41,23 @@
 DEFAULT_SPAN_RETENTION_DAYS = 90
 
 
-def _build_snuba_span(relay_span: Mapping[str, Any]) -> MutableMapping[str, Any]:
-    project = Project.objects.get_from_cache(id=relay_span["project_id"])
+@lru_cache(maxsize=10000)
+def get_organization(project_id: int) -> Tuple[Organization, int]:
+    project = Project.objects.get_from_cache(id=project_id)
     organization = project.organization
     retention_days = (
         quotas.backend.get_event_retention(
             organization=organization,
         )
         or DEFAULT_SPAN_RETENTION_DAYS
     )
+    return organization, retention_days
```

Comment:
do you need to return the whole org and keep it all in cache? seems like only the ID is used

Reply:
Indeed.